### PR TITLE
Refactor entrypoint `auth_ejabberd.php`

### PR DESCRIPTION
**Pull request by Art4**
This PR refactors the entrypoint `auth_ejabberd.php` by moving most of the logic inside the App class.

Requires:

- [x] #14620

### CODE REVIEW

**Comment by MrPetovan** [comment]
https://github.com/friendica/friendica/pull/14620 has been merged, is this ready to review?

File: bin/auth_ejabberd.php [hunk 1]
```diff
@@ -45,43 +45,13 @@
...
-}
-
-$directory = dirname($_SERVER["argv"][0]);
-
-if (substr($directory, 0, 1) != DIRECTORY_SEPARATOR) {
-	$directory = $_SERVER["PWD"] . DIRECTORY_SEPARATOR . $directory;
-}
-
-$directory = realpath($directory . DIRECTORY_SEPARATOR . "..");
+chdir(dirname(__FILE__, 2));
```
Comment (MrPetovan) [inline bin/auth_ejabberd.php:49]: `dirname(__FILE__, 2)` is used 3 times in this file, wouldn't it be a good idea to make it into a variable?

Comment (Art4) [reply bin/auth_ejabberd.php:49]: I'd say no. I would like to keep the global namespace as clean a possible. And the Dice rules file will be refactored in the future again. Personally I think keeping the hardcoded `dirname(__FILE__, 2)` in the entry points increases the readability. But we can replace it with `dirname(__DIR__, 1)`. What do you think?

Comment (MrPetovan) [reply bin/auth_ejabberd.php:49]: Personally, I prefer using `__DIR__` if it's used directly, meaning as a replacement for `dirname(__FILE__)` specifically. If we need to go up more folder levels, then I prefer using `dirname(__FILE__, 2+)`, I feel it is more intuitive file-tree wise.

File: src/App.php [hunk 2]
```diff
@@ -156,18 +166,40 @@ public function processRequest(ServerRequestInterface $request, float $start_tim
...
 
 		$this->container = $this->container->addRules($addonLoader->getActiveAddonConfig('dependencies'));
-		$this->container = $this->container->addRule(Mode::class, [
-			'call' => [
-				['determineRunMode', [false, $request->getServerParams()], Dice::CHAIN_CALL],
-			],
-		]);
+	}
 
+	private function setupLegacyServerLocator(): void
```
Comment (MrPetovan) [inline src/App.php]: Shouldn't be this named `setupLegacyServiceLocator`, related to the conversation we've had recently about the Service Locator anti-pattern?
```suggestion
	private function setupLegacyServiceLocator(): void
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/App.php (hunk 2 above)
Comment (Art4) [reply src/App.php]: You are right, fixed.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**